lib/server-starter.js

```diff
@@ -86,7 +90,23 @@ class Server extends EventEmitter {
       this.emit('exit', code, signal);
     });
 
-    return new Promise(resolve => this._srv.close(resolve));
+    return new Promise(resolve => this._srv.close(
+      () => {
+        if (this._useFdPass) resolve();
+        else {
+          const now = new Date();
+          const timeToStop = new Date(now.getTime() + connectTimeout);
+          const port = this.port;
+          (function loop () {
+            const connection = net.connect(port, resolve);
+            connection.on('error', err => {
+              if (err.code === 'ECONNREFUSED' && new Date() < timeToStop) setTimeout(loop, retryTime);
+              else resolve(); // this is intented: don't reject, just stop delaying
```

Comment:
One of my requirements was that it rejects on errors. Are you sure you don't just want to make your own module?